### PR TITLE
fix(taskframework): handle all preparing and running task to failed when task-framework be set disabled

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
module-taskframework
#### What this PR does / why we need it:
1. we should handle all running and failed jobs to failed when task-framework be set disabled

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 1]
```diff
@@ -154,6 +154,14 @@ public Page<JobEntity> findHeartTimeTimeoutJobs(int timeoutSeconds, int page, in
         return page(condition, page, size);
     }
 
+    @Override
+    public Page<JobEntity> findNotTerminalJobs(int page, int size) {
+        Specification<JobEntity> condition = Specification.where(getRecentDaySpec(RECENT_DAY))
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:159]: Only querying the data of the past 30 days may miss published tasks, which may be optimized in the future.

**Review by guowl3 — APPROVED** [review]

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:159]: ok, this wil be optimized by archive history data

**Review by yhilmare — DISMISSED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 2]
```diff
@@ -154,6 +154,14 @@ public Page<JobEntity> findHeartTimeTimeoutJobs(int timeoutSeconds, int page, in
         return page(condition, page, size);
     }
 
+    @Override
+    public Page<JobEntity> findNotTerminalJobs(int page, int size) {
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java]: okey,  looks more reasonable

**Review by guowl3 — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**